core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -4004,6 +4004,19 @@ public static long unixDateExtract(String rangeString, long date) {
    * {@code FORMAT_DATETIME}, {@code FORMAT_TIME}, {@code TO_CHAR} functions. */
   @Deterministic
   public static class DateFormatFunction {
+    // Timezone to use for PostgreSQL parsing of timestamps
+    private static final ZoneId LOCAL_ZONE;
+    static {
+      ZoneId zoneId;
+      try {
+        // Currently the parsed timestamps are expected to be in UTC
```

Comment:
The value returned from the existing `to_timestamp` is expected to be the millis from epoch in UTC.
https://github.com/apache/calcite/blob/main/core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java#L4065

At some point in the future, this could be changed to support timezones, since you could do something like:
```
to_timestamp('2024-06-03 15:32:46+03:30')
```

Reply:
Please add this information as comments in the code.

Reply:
Improved the comment here.